biorxiv/corpora_comparison/biorxiv_pmc_figure_generator.py

```diff
@@ -0,0 +1,486 @@
+            f"<b>{row.lemma}</b><br />Preprint:{row.preprint_count:,}"
+            f"<br />Published:{row.published_count:,}<br />"
+            f"Odds Ratio:{row.odds_ratio:.2f}"
+        ),
+        axis=1
+    ))
+    
+)
+plot_df.head()
+
+
+# In[17]:
+
+
+fig=go.Figure(
+    go.Treemap(
+        branchvalues="remainder",
+        labels = ["preprint","published"] + plot_df.label.tolist(),
+        parents = ["", ""] + plot_df.repository.tolist(),
+        values = [0,0] + plot_df.odds_ratio.tolist(),
+        textinfo = "label",
+        textfont = {
+            'size':15
+        }
+    )
+)
+fig.update_layout(
+    width=1280, 
+    height=720,
+    treemapcolorway=["#a6cee3", "#b2df8a"],
+    title={
+        'text':f"Comparison of Top {plot_df.shape[0]} Words between Preprints and Published Papers",
+        'y':0.9,
+        'x':0.5,
+        'xanchor': 'center',
+        'yanchor': 'top'}
+)
+fig.update_layout(uniformtext=dict(minsize=10))
+fig.show()
+
+
+# In[18]:
+
+
+fig.write_image("output/svg_files/preprint_published_comparison_treemap.svg", scale=2.5)
+
+
+# ## Line Plot
+
+# In[19]:
+
+
+full_plot_df = (
+    preprint_published_comparison
+    .assign(
+        lower_odds=lambda x: np.exp(
+            np.log(x.odds_ratio) - 1.96*(
+                # log(odds) - z_alpha/2*sqrt(1/a+1/b+1/c+1/d)
+                np.sqrt(
+                    1/x.preprint_count + 1/x.published_count + 1/x.preprint_total + 1/x.published_total
+                )
+            )
+        ),
+        upper_odds=lambda x: np.exp(
+            np.log(x.odds_ratio) + 1.96*(
+                # log(odds)+ z_alpha/2*sqrt(1/a+1/b+1/c+1/d)
+                np.sqrt(
+                    1/x.preprint_count + 1/x.published_count + 1/x.preprint_total + 1/x.published_total
+                )
+            )
+        )
+    )
+)
+full_plot_df.head()
+
+
+# In[20]:
+
+
+plot_df = (
+    full_plot_df
+    .sort_values("odds_ratio", ascending=False)
+    .head(subset)
+    .append(
+        full_plot_df
+        .sort_values("odds_ratio", ascending=False)
+        .tail(subset)
+    )
+)
+plot_df.head()
+
+
+# In[21]:
+
+
+g = (
+    p9.ggplot(plot_df, p9.aes(x="lemma", y="odds_ratio"))
+    + p9.geom_pointrange(p9.aes(ymin="lower_odds", ymax="upper_odds"), position=p9.position_dodge(width=5))
+    + p9.scale_x_discrete(limits=plot_df.sort_values("odds_ratio", ascending=True).lemma.tolist())
+    + p9.geom_hline(p9.aes(yintercept=1), linetype = '--', color='grey')
+    + p9.coord_flip()
+    + p9.theme_seaborn(context='paper')
+    + p9.theme(
+        # 1024, 768
+        figure_size=(13.653333333333334, 10.24),
+        axis_text_y=p9.element_text(family='DejaVu Sans', size=12),
+        panel_grid_minor=p9.element_blank(),
+        axis_title=p9.element_text(size=15),
+        axis_text_x=p9.element_text(size=11, weight="bold")
+    )
+    + p9.labs(x=None)
+)
+g.save("output/svg_files/preprint_published_frequency_odds.svg", dpi=75)
+print(g)
```

Comment:
It's interesting that the long tail in this lineplot is on the <1 side while it was on the >1 side in the PMC vs BioRxiv line plot. Is there a reason for that? Or just random outliers?

Reply:
>It's interesting that the long tail in this lineplot is on the <1 side while it was on the >1 side in the PMC vs BioRxiv line plot. Is there a reason for that? Or just random outliers?

Good catch. Honestly, I think it is just a coincidence. Words are sorted by their frequency, which just happens to fall into this pattern.